### PR TITLE
Faster bitmap marking

### PR DESCRIPTION
**Pull request by KeenS**
This is further work of #304 .Would be faster and up to date to master.

This patch depends `posix_memalign` or similar functionality so depends on libc. However, you can select gc algorithms at compile time.

# TODO
* [x] better allocate/mark/sweep algorithm optimized to bitmap
* [x] cleaner configration

### CODE REVIEW

**Comment by KeenS** [comment]
I'll give you some my experiment result

* this path makes picrin faster (in `make test-logic` , 8.4s -> 6.4s on my laptop)
* Using SSE 4.2's `popcnt` as implementation of `numofbits` resulted in slow gc time
* Using `uint64_t` instead of `uint32_t` as bitmap resulted in slow gc time
* Making bitmap index to bitmap tree resulted in slow gc time. This is related to heap page size. If the heap page size is bigger than current value, bitmap tree would be faster
* Implementing sticky mark generational gc was difficult, so I didn't (sticky mark itself is easy, but implementing remember set is difficult)

**Comment by KeenS** [comment]
@nyuichi Please review and merge if this looks good to you

File: extlib/benz/read.c [hunk 1]
```diff
@@ -640,7 +640,7 @@ read_nullable(pic_state *pic, xFILE *file, int c, struct reader_control *p)
 }
 
 static pic_value
-read(pic_state *pic, xFILE *file, int c, struct reader_control *p)
```
Comment (nyuichi) [inline extlib/benz/read.c]: Any reasons to change the name of read function? @KeenS

Comment (KeenS) [inline extlib/benz/read.c]: I included `unistd.h` to use `posix_memalign`, which causes name conflict against `read` and write`.

Comment (nyuichi) [inline extlib/benz/read.c]: @KeenS 

Oh, I see. That makes sense.

**Pull request merged**